### PR TITLE
All claims 31 5 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds a summary of evidence page following the evidence upload pages in the All Claims form.
A list for each evidence type uploaded is shown.

NOTE: This PR doesn't include summary for fetching private records, since that's part of 4142 and hasn't been built yet. Once the 4142 is integrated into the All Claims form, we'll probably want to circle back and add a summary view for private medical records that need to be fetched from a facility rather than uploaded directly.

## Testing done
- Local testing
- Unit tests

## Screenshots
No Evidence:
screen shot 2018-10-04 at 9 45 24 am

-----

Some evidence types but not all (VA & Private but no lay evidence):
screen shot 2018-10-04 at 9 26 08 am

## Acceptance criteria
- [x] Veteran sees a list of uploaded / entered evidence for each type of evidence entered
- [x] Veteran sees a warning when no evidence submitted / uploaded

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by peggygannon — APPROVED** [review]
@markgreenburg, Hi, Mark. Screenshots look good. 

My only comment is that under "We'll get your VA medical records from":
the bullet listed says "Private medical facility"..

 I just want to make sure that a VA medical center or hospital will be pulled in there.

**Comment by markgreenburg** [comment]
@peggygannon yep, that's just random autocomplete data I used to fill in the VA facility info. Sorry for the confusion.

**Review by cvalarida — APPROVED** [review]
Beautiful! I think the tests could use one more check, but it looks great.

File: src/applications/disability-benefits/all-claims/tests/pages/summaryOfEvidence.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          'view:hasEvidence': true,
+          vaTreatmentFacilities: [
+            { treatmentCenterName: 'Sommerset' },
+            { treatmentCenterName: 'Huntsville' },
+          ],
+        }}
+      />,
+    );
+
+    expect(form.find('li').length).to.equal(2);
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/pages/summaryOfEvidence.unit.spec.jsx]: Good call!

File: src/applications/disability-benefits/all-claims/content/summaryOfEvidence.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import _ from '../../../../platform/utilities/data';
+
+export const summaryOfEvidenceDescription = ({ formData }) => {
+  const vaEvidence = _.get('vaTreatmentFacilities', formData, []);
+  const privateEvidence = _.get('privateMedicalRecords', formData, []);
+  const layEvidence = _.get('additionalDocuments', formData, []);
+  const hasEvidence = !!vaEvidence.concat(privateEvidence, layEvidence).length;
+  const selectedEvidence = _.get('view:hasEvidence', formData, false);
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/content/summaryOfEvidence.jsx]: Yeah, I gave up on that one. Monday morning seems to be yielding fresher ideas, though.

**Pull request merged**